src/django_rich/test.py

```diff
@@ -42,6 +46,8 @@ class RichTextTestResult(unittest.TextTestResult):
     # https://github.com/python/typeshed/pull/7340
     showAll: bool
 
+    collectedDurations: list[tuple[TestCase, float]] = []
```

Comment:
this data structure works but there are a couple improvements...

first, we don't want to keep around `TestCase` instances when we only need the `str()` later - I'm worried about state on test case instances taking up excessive memory.

second, we could maintain the list in sorted order, by using [`heapq`](https://docs.python.org/3.10/library/heapq.html) or similar. then we could keep just the top N items during the run, rather than storing timing for *all* tests only to output a limited count later

also I know unittest uses camel case but I'm tempted to use normal snake_case for custom attributes

Reply:
I think I'm wrong about point 1 here - unittest already has the list of tests in the suite class, so we wouldn't be making them live any longer

We could actually save a little time then not converting to string until required